### DISCUSSION TITLE
How to delete/uninstall Teleport from a server

### DISCUSSION THREAD

**Discussion by dinardalvi** [gravitational/teleport#9920] [Q&A] [answered]
I am trying teleport and how do I delete the added server from the WebUI? 

Thanks

**Comment by webvictim** [comment, marked as answer]
Here's some instructions on how to remove a server added using the "Add Server" button in the Teleport UI.

1) Stop any running Teleport processes (`sudo killall teleport`)

If you have Teleport running via systemd on Linux, you should also run:

- `sudo systemctl stop teleport`
- `sudo systemctl disable teleport`

If you ran this script on a Mac, follow these additional steps to stop the Teleport agent:
- `sudo launchctl unload /Library/LaunchDaemons/com.goteleport.teleport.plist`
- `sudo rm -f /Library/LaunchDaemons/com.goteleport.teleport.plist`

2) Remove the Teleport package/binaries. This method varies depending on what system you're running on - the install script will tell you which method it uses to install in its output when running.

If you installed using an RPM package:
- `sudo rpm -e teleport`

If you installed using yum:
- `sudo yum -y remove teleport`

If you installed using dnf:
- `sudo dnf -y remove teleport`

If you installed using a DEB package:
- `sudo dpkg -r teleport`

If you installed using apt:
- `sudo apt -y remove teleport`

If you installed from a tarball:
- `sudo rm -f /usr/local/bin/teleport`
- `sudo rm -f /usr/local/bin/tctl`
- `sudo rm -f /usr/local/bin/tsh`
- `sudo rm -f /usr/local/bin/tbot`
- `sudo rm -f /usr/local/bin/fdpass-teleport`

If you are running Teleport on Linux, you should also remove the `systemd` unit file:
- `sudo rm -f /etc/systemd/system/teleport.service`

3) Remove any remaining Teleport data
- remove the /var/lib/teleport directory (`sudo rm -rf /var/lib/teleport`)
- remove the /etc/teleport.yaml config file (`sudo rm -f /etc/teleport.yaml`)

At this point Teleport is gone completely. Any servers you added this way will disappear from the Teleport web UI within 10-15 minutes. Assuming you copy/pasted the `sudo bash -c...` command from the web UI, the `install.sh` script ran in memory so does not need to be cleaned up separately.

**Reply by slhck** [reply]
This didn't work for me under Ubuntu. I also had to run:

```
systemctl stop teleport
systemctl disable teleport
systemctl daemon-reload
```

to get rid of the remaining systemd entries.

**Reply by webvictim** [reply]
Thanks, updated the steps.

**Comment by rob-scheepens** [comment]
How would I remove any remaining binaries on Mac?

**Reply by webvictim** [reply]
Use the `launchctl` commands above and then remove the files under `/usr/local/bin` as described.

**Reply by codevalve** [reply]
I used this from the accepted answer for my MacOS install : 

```
sudo rm -f /usr/local/bin/teleport
sudo rm -f /usr/local/bin/tctl
sudo rm -f /usr/local/bin/tsh

sudo rm -rf /var/lib/teleport
sudo rm -f /etc/teleport.yaml
```

**Reply by KurtLehnardt** [reply]
This worked for me to uninstall teleport after I installed it from https://cdn.teleport.dev/tsh-<version>.pkg

> I used this from the accepted answer for my MacOS install :
> 
> ```
> sudo rm -f /usr/local/bin/teleport
> sudo rm -f /usr/local/bin/tctl
> sudo rm -f /usr/local/bin/tsh
> 
> sudo rm -rf /var/lib/teleport
> sudo rm -f /etc/teleport.yaml
> ```
